core/deployment/src/main/java/io/quarkus/deployment/QuarkusAugmentor.java

```diff
@@ -64,6 +66,7 @@
         this.excludedFromIndexing = builder.excludedFromIndexing;
         this.liveReloadBuildItem = builder.liveReloadState;
         this.buildSystemProperties = builder.buildSystemProperties;
+        this.moduleDir = builder.moduleDir;
```

Comment:
We really don't want this, as it makes assumptions that the augment process only happens from within the scope of a project, which may not always be the case in the future.

Reply:
Don't you just need the target directory rather than the whole project?

If you are doing this to get the dockerfile then I think it would actually make sense to move the dockerfile to src/main/resources so it ends up in the final archive. This means that it would still be possible to re-run augmentation without the project being present.

Reply:
No, target directory won't do.
Also, having the path to dockerfiles won't do either (although it would be nice to have).

The quarkus dockerfiles assume that the build is executed from within the module dir. Why?

The following piece of code is taken from `Dockerfile.jvm`:

    COPY target/lib/* /deployments/lib/
    COPY target/*-runner.jar /deployments/app.jar

The instructions above only make sense from within the module dir. If the build is executed with any other path, it will just fail.

So, if we want to be able to trigger a docker build, we do need to know this directory.

Now, a potential alternative would be to modify the dockerfiles on the fly, to remove any reference to `target/` and then just use the `targetDir` for the build, which IMHO is more hacky than just adding the `moduleDir` to the Augmentor.

Reply:
The problem is that there might not be a module dir, we want to be able to augment standalone jar files as well. We are likely going to need to modify our Dockerfile for this use case.

Reply:
Modify the Dockerfiles provided by quarkus will result in ugly Dockerfiles. I will try and see if I can modify them on demand, right before the build. I'll update the pull request shortly

Reply:
I mean't that we would probably need to look at producing different docker files in our quickstarts. Possibly we could even include the generic dockerfile in the extension and use it so the user does not have to do anything at all, and give the user the ability to customise it. 

This probably needs to be discussed with a wider audience.

Reply:
I'll start working towards this direction.

Reply:
This was completely re-written, we are now using the Dockerfiles under `src/main/docker` or from any user provided location.

Detection of `src/main/docker`is done by recursively navigating upwards from the output folder, till `src/main/docker` exists.